mirrord/layer/src/common.rs

```diff
@@ -11,29 +12,77 @@ use crate::{
     HOOK_SENDER,
 };
 
+/// Type alias for a queue of responses from the agent, where these responses are [`RemoteResult`]s.
+///
+/// ## Usage
+///
+/// Most mirrord features that interact with the agent use this type to keep a queue of the
+/// operations, keeping them in order, and allowing us to match a hook request to a response. It's
+/// done in this way due to us not having identifiers for the hook requests.
+///
+/// - The usual flow is:
+///
+/// 1. `push_back` the [`oneshot::Sender`] that will be used to produce the [`RemoteResult`];
+///
+/// 2. When the operation gets a response from the agent:
+///     1. `pop_front` to get the [`oneshot::Sender`], then;
+///     2. `Sender::send` the result back to the operation that initiated the request.
 pub(crate) type ResponseDeque<T> = VecDeque<ResponseChannel<T>>;
 
+/// Sends a [`HookMessage`] through the global [`HOOK_SENDER`].
```

Comment:
Is `HOOK_SENDER` going to be documented separately?

Reply:
Not sure what you mean by that? 

We need to add docs to everything in order to turn on "deny missing docs".

Reply:
I mean, saying that it sends a message through HOOK_SENDER doesn't add much info if we don't know what HOOK_SENDER does (i.e. does it send messages from the hooks? to the hooks?)

Reply:
I think it's fine to rely on the reader navigating to HOOK_SENDER to read more about it (that's why I initially asked if it's going to be documented)